base/linalg/adjtrans.jl

```diff
@@ -136,30 +136,15 @@ hcat(tvs::Transpose{T,Vector{T}}...) where {T} = _transpose_hcat(tvs...)
 ### higher order functions
 # preserve Adjoint/Transpose wrapper around vectors
 # to retain the associated semantics post-map/broadcast
-
-# vectorfy takes an Adoint/Transpose-wrapped vector and builds
-# an unwrapped vector with the entrywise-same contents
-vectorfy(x::Number) = x
-vectorfy(adjvec::AdjointAbsVec) = map(Adjoint, adjvec.parent)
-vectorfy(transvec::TransposeAbsVec) = map(Transpose, transvec.parent)
-vectorfyall(transformedvecs...) = (map(vectorfy, transformedvecs)...,)
-
-# map over collections of Adjoint/Transpose-wrapped vectors
-# note that the caller's operation `f` should be applied to the entries of the wrapped
-# vectors, rather than the entires of the wrapped vector's parents. so first we use vectorfy
-# to build unwrapped vectors with entrywise-same contents as the wrapped input vectors.
-# then we map the caller's operation over that set of unwrapped vectors. but now re-wrapping
-# the resulting vector would inappropriately transform the result vector's entries. so
-# instead of simply mapping the caller's operation over the set of unwrapped vectors,
-# we map Adjoint/Transpose composed with the caller's operationt over the set of unwrapped
-# vectors. then re-wrapping the result vector yields a wrapped vector with the correct entries.
-map(f, avs::AdjointAbsVec...) = Adjoint(map(Adjoint∘f, vectorfyall(avs...)...))
-map(f, tvs::TransposeAbsVec...) = Transpose(map(Transpose∘f, vectorfyall(tvs...)...))
-
-# broadcast over collections of Adjoint/Transpose-wrapped vectors and numbers
-# similar explanation for these definitions as for map above
-broadcast(f, avs::Union{Number,AdjointAbsVec}...) = Adjoint(broadcast(Adjoint∘f, vectorfyall(avs...)...))
-broadcast(f, tvs::Union{Number,TransposeAbsVec}...) = Transpose(broadcast(Transpose∘f, vectorfyall(tvs...) ...))
+#
+# note that the caller's operation f operates in the domain of the wrapped vectors' entries.
+# hence the Adjoint->f->Adjoint shenanigans applied to the parent vectors' entries.
+map(f, avs::AdjointAbsVec...) = Adjoint(map((xs...) -> Adjoint(f(Adjoint.(xs)...)), parent.(avs)...))
```

Comment:
given recent discussions shouldn't this be `adjoint.(xs)...`?

Reply:
Absolutely, given recent discussion these calls should eventually become `adjoint`/`transpose` :). That said, with master as it stands, `Adjoint`/`Transpose` are correct, whereas I am not entirely confident that `adjoint`/`transpose` are correct and will remain correct through the few remaining pull requests in the #5332 series. After those remaining pull requests in the #5332 series, changing these calls to `adjoint`/`transpose` should become correct with certainty. Thoughts? :) Thanks!

Reply:
As long as this is an intermediate state, I'm fine with it.

Reply:
With a bit more thought, writing these methods in `Adjoint`/`Transpose` may also be ultimately best: The relevant objects are `<:AbstractVector`, and as such `Adjoint`/`Transpose`-wrapping is correct, precise, and unambiguous  (whereas `adjoint`/`transpose` will be more nebulous). Best!